draft-ietf-taps-arch.md

```diff
@@ -216,35 +216,53 @@ Transport Services implementations are meant to be flexible at connection establ
 
 Flexibility after connection establishment is also important. Transport protocols that can migrate between multiple network-layer interfaces need to be able to process and react to interface changes. Protocols that support multiple application-layer streams need to support initiating and receiving new streams using existing connections.
 
-# Design Principles {#design}
+# API and Implementation Requirements {#requirements}
 
 The goal of the Transport Services architecture is to redefine the interface between applications and transports in a way that allows the transport layer to evolve and improve without fundamentally changing the contract with the application. This requires a careful consideration of how to expose the capabilities of protocols.
 
-There are several degrees in which a Transport Services system is intended to offer flexibility to an application: it can provide access to multiple sets of protocols and protocol features; it can use these protocols across multiple paths that could have different performance and functional characteristics; and it can communicate with different remote systems to optimize performance, robustness to failure, or some other metric. Beyond these, if the API for the system remains the same over time, new protocols and features could be added to the system's implementation without requiring changes in applications for adoption.
+There are several degrees in which a Transport Services system is intended to offer flexibility to an application: it can provide access to multiple sets of protocols and protocol features; it can use these protocols across multiple paths that could have different performance and functional characteristics; and it can communicate with different remote systems to optimize performance, robustness to failure, or some other metric. Beyond these, if the API for the system remains the same over time, new protocols and features can be added to the system's implementation without requiring changes in applications for adoption.
 
-## Common APIs for Common Features
+In order to provide this functionality without causing incompatibility or introducing security vulnerabilities, Transport Services APIs and Implementations should follow the set of requirements described here.
```

Comment:
Interim: Add a comment that this is the only normative section of the document

Reply:
The subsection with requirements could also be called out in the overview as one sentence in 1.2?

Reply:
Added